sgkit_vcf/tbi.py

```diff
@@ -0,0 +1,117 @@
+import gzip
+from dataclasses import dataclass
+from typing import Any, Sequence
+
+import numpy as np
+
+from sgkit.typing import PathType
+from sgkit_vcf.utils import at_eof, get_file_offset, read_bytes
+
+TABIX_LINEAR_INDEX_INTERVAL_SIZE = 1 << 14  # 16kb interval size
+
+
+@dataclass
+class Header:
+    n_ref: int
+    format: int
+    col_seq: int
+    col_beg: int
+    col_end: int
+    meta: int
+    skip: int
+    l_nm: int
+
+
+@dataclass
+class Chunk:
+    cnk_beg: int
+    cnk_end: int
+
+
+@dataclass
+class Bin:
+    bin: int
+    chunks: Sequence[Chunk]
+
+
+@dataclass
+class TabixIndex:
```

Comment:
Does it cause issues if there are multiple variants with the same contig + position (i.e. if they were split)?

Reply:
The index will always return the file offset of the first record for each linear index interval. There is a potential problem when querying the VCF by region since it will return all records that overlap that region. This is solved by filtering out any that don't start in the region (see `region_filter`) since ones with earlier start positions are handled by the previous region.

Is this the case you're thinking about or is there another one?

Reply:
Sounds good if the filter would either let them all through or skip them all.  The placement for this comment was somewhat arbitrary -- I just wanted to confirm that nothing in the partitioning or indexing logic assumed single rows for any one contig + position combination.

Reply:
No, there are no cases that I know of.